llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -1426,6 +1468,15 @@ void RISCVInsertVSETVLI::emitVSETVLIs(MachineBasicBlock &MBB) {
       PrefixTransparent = false;
     }
 
+    if (isVectorCopy(ST->getRegisterInfo(), MI)) {
+      if (!PrevInfo.isCompatible(DemandedFields::all(), CurInfo, LIS)) {
+        insertVSETVLI(MBB, MI, MI.getDebugLoc(), CurInfo, PrevInfo);
```

Comment:
Add a statistics to record how many extra vsetivli we have inserted?

Reply:
I'm not sure if it's useful to record the number of vsetvlis inserted specifically for copies since a lot of them will be coalesced away. I can try and get the diff in the total number of vsetvlis emitted (NumInsertedVSETVL - NumCoalescedVSETVL)